### PR TITLE
add: debug logs for timeout

### PR DESCRIPTION
**Pull request by 07souravkunda**
Add more debugging logs for o11y timeouts.

### CODE REVIEW

**Review by akanksha1909 — DISMISSED** [review]

**Comment by sauravdas1997** [comment]
Let's add logs for eventlistener on hooks as well, to identify if any hooks are getting missed. Example: `EVENT_HOOK_END`

File: bin/testObservability/helper/helper.js [hunk 1]
```diff
@@ -475,14 +484,18 @@ exports.batchAndPostEvents = async (eventUrl, kind, data) => {
...
 
   try {
+    const eventsUuids = data.map(eventData => `${eventData.event_type}:${eventData.test_run ? eventData.test_run.uuid : (eventData.hook_run ? eventData.hook_run.uuid : null)}`).join(', ');
+    exports.debugOnConsole(`[Request Batch Send] for events:uuids ${eventsUuids}`);
     const response = await nodeRequest('POST',eventUrl,data,config);
+    exports.debugOnConsole(`[Request Batch Response] for events:uuids ${eventsUuids}`);
     if(response.data.error) {
       throw({message: response.data.error});
     } else {
       exports.debug(`${kind} event successfull!`)
```
Comment (sauravdas1997) [inline bin/testObservability/helper/helper.js:494]: let's add a log here as well, if success

File: bin/testObservability/helper/helper.js [hunk 2]
```diff
@@ -522,6 +535,7 @@ exports.uploadEventData = async (eventData, run=0) => {
       
       exports.requestQueueHandler.start();
       const { shouldProceed, proceedWithData, proceedWithUrl } = exports.requestQueueHandler.add(eventData);
+      exports.debugOnConsole(`[Request Queue] ${eventData.event_type} with uuid ${eventData.test_run ? eventData.test_run.uuid : (eventData.hook_run ? eventData.hook_run.uuid : null)} is added`)
```
Comment (sauravdas1997) [inline bin/testObservability/helper/helper.js:538]: there is a return statement at the beginning of the method as well. Let's add a log at the start of `uploadEventData`

**Comment by sauravdas1997** [comment]
Log line [here](https://github.com/browserstack/browserstack-cypress-cli/pull/863/files#diff-4798b5b56abaf7e213859baef6ab90f8a654b138bb5d89c470a0384febf91d66R489) and [here](https://github.com/browserstack/browserstack-cypress-cli/pull/863/files#diff-4798b5b56abaf7e213859baef6ab90f8a654b138bb5d89c470a0384febf91d66R393) to confirm that there are no silent errors, because we won't be able to see these log lines though there are debug logs

File: bin/testObservability/helper/helper.js (hunk 1 above)
Comment (07souravkunda) [reply bin/testObservability/helper/helper.js:494]: Logging the complete response. If there is no error then it's successful. Do we need a redundant log?

**Review by sauravdas1997 — APPROVED** [review]

**Review by Baaryan — APPROVED** [review]

**Pull request merged**